core/src/main/java/org/geysermc/geyser/session/cache/LodestoneCache.java

```diff
@@ -54,6 +55,7 @@ public final class LodestoneCache {
 
     public void cacheInventoryItem(GeyserItemStack itemStack) {
         CompoundTag tag = itemStack.getNbt();
+        Objects.requireNonNull(tag);
```

Comment:
Considering this method just returns if `tag.get("LodestonePos")` returns null, then I would simply return if `tag` is null as well.